dali/kernels/reduce/reduce_gpu_impl.cuh

```diff
@@ -258,8 +261,8 @@ auto GetPreprocessorsHelper(std::true_type, const ReduceImpl *impl, WorkArea *wa
 template <int non_reduced_dims, typename ReduceImpl>
 auto GetPreprocessorBanksHelper(
       std::true_type, const ReduceImpl *impl, WorkArea *wa, int reduced_axis)
-      ->decltype(impl->template GetPreprocessorBanksImpl<non_reduced_dims>(*wa, reduced_axis)) {
-  return impl->template GetPreprocessorBanksImpl<non_reduced_dims>(*wa, reduced_axis);
+      ->decltype(impl->GetPreprocessorBanksImpl(*wa, reduced_axis, int_const<non_reduced_dims>())) {
```

Comment:
That solves the CUDA 9 bug.

Reply:
So yo can add some comment in the code.

Reply:
It doesn't make much sense to comment it in the code - how would I do it? Show an example of (arguably more complicated) code that _didn't_ work?

Reply:
State that this is the fix for CUDA 9 compiler bug and give some background. So when we deprecate CUDA 9 we can remove this WAR.

Reply:
But I think it's actually better this way! It's not a workaround.